reference/04-Patterns/block-bindings-api.md

```diff
@@ -0,0 +1,242 @@
+
+## How Synced Patterns leverage the API
+
+If we open up the **Edit Original** screen of our Synced Pattern and choose the code editor view **(⇧⌥⌘M
+)**, we can get a little more insight into what's happening to make this work.
+
+Let's take a closer look at our Heading block within our pattern and format it a bit for readability.
+
+```html
+<!--
+wp:heading
+{
+  "metadata": {
+    "bindings": {
+      "__default": {
+        "source": "core/pattern-overrides"
+      }
+    },
+    "name": "Call to Action Title"
+  },
+  "fontSize":"x-large"
+}
+-->
+<h2 class="wp-block-heading has-x-large-font-size">Want to get in touch?</h2>
+<!-- /wp:heading -->
+```
+
+Here we can see our new attribute added to the block for **"binding"** that controls the synchronization.  In a Synced Pattern, the binded attribute is set as **"__default"**, which essentially means it's bound to all default supported attributes we initially set in our pattern.  In our Heading's case, that would mean the *content* attribute but for a Button it would mean *url*, *text*, *linkTarget*, and *rel*. We can also see it has a source set to **"core/pattern-overrides"**.  It is these attributes that we can alter using the Block Bindings API to create our own logic for managing our patterns content.
+
+## Creating our own Bindings
+
+Let's say we are using our pattern on a post type about products where the individual items data is stored in various post meta fields.  Using the Block Bindings API, we can set defaults and use those meta fields to dynamically alter our patterns content.
+
+Let's quickly look at the structure of the attributes we will need to use on our blocks.  It's very similar to the one used on a Synced Pattern, with a notable difference in using our custom source (which we will create below) and then adding a key within the **args** object which allows for dynamic content.
+
+```json
+{
+  "metadata": {
+    "bindings": {
+      "content": {
+        "source": "tenup/cta-binding",
+        "args": {
+          "key": "cta_heading"
+        }
+      }
+    }
+  }
+}
+```
+
+Here, we're using a Heading block as an example where we are binding its **content** attribute to our custom source.
+
+:::caution
+Much like Synced Pattern Overrides, when using the Block Bindings API it is very important to know that our blocks are not optional, and we cannot display them conditionally.
+
+For example, if we bind our Heading's `content` attribute to post meta, but that post meta does not return a value, an empty `<h2>` tag will display on the frontend.  Similarly, we cannot use multiple Buttons in our pattern and have the ability to sometimes display just a single one.  Whatever blocks are given bindings, their markup will **always** be returned.
+
+Because of this, when creating your pattern and setting up your custom binding source, it is recommended to always set default values.  More on that below.
```

Comment:
I'd also like to mention the issue of orphaned labels somewhere

Reply:
I'm also not sure what this one means, can you clarify please?

Reply:
Ahh of course. 

What I mean is because there is no optional logic you can end up with labels for a meta value that still displays even if the meta value is empty. 

Where as in PHP we would have wrapped the entire thing in an if statement and just not rendered the label/title if the value doesn't exist.

Reply:
Done!